Sources/PrimerSDK/Classes/Data Models/Theme/Public/PublicThemeData.swift

```diff
@@ -0,0 +1,158 @@
+    public var buttons: ButtonStyleData
+    public var input: InputThemeData
+    
+    public init(
+        colors: ColorSwatchData = ColorSwatchData(),
+        view: ViewThemeData = ViewThemeData(),
+        text: TextStyleData = TextStyleData(),
+        buttons: ButtonStyleData = ButtonStyleData(),
+        input: InputThemeData = InputThemeData()
+    ) {
+        self.colors = colors
+        self.view = view
+        self.text = text
+        self.buttons = buttons
+        self.input = input
+    }
+}
+
+public class ButtonStyleData {
+    public var main: ButtonThemeData
+    public var paymentMethod: ButtonThemeData
+    
+    public init(
+        main: ButtonThemeData = ButtonThemeData(),
+        paymentMethod: ButtonThemeData = ButtonThemeData()
+    ) {
+        self.main = main
+        self.paymentMethod = paymentMethod
+    }
+}
+
+public class ButtonThemeData {
+    public var defaultColor: UIColor?
+    public var disabledColor: UIColor?
+    public var errorColor: UIColor?
+    public var cornerRadius: CGFloat?
+    public var text: TextThemeData
+    public var border: BorderThemeData
+
+    public init(
+        defaultColor: UIColor? = nil,
+        disabledColor: UIColor? = nil,
+        errorColor: UIColor? = nil,
+        cornerRadius: CGFloat? = nil,
+        text: TextThemeData = TextThemeData(),
+        border: BorderThemeData = BorderThemeData()
+    ) {
+        self.defaultColor = defaultColor
+        self.disabledColor = disabledColor
+        self.errorColor = errorColor
+        self.text = text
+        self.border = border
+        self.cornerRadius = cornerRadius
+    }
+}
+
+public class ViewThemeData {
+    public var backgroundColor: UIColor?
+    public var cornerRadius: CGFloat?
+    public var safeMargin: CGFloat?
+    
+    public init(
+        backgroundColor: UIColor? = nil,
+        cornerRadius: CGFloat? = nil,
+        safeMargin: CGFloat? = nil
+    ) {
+        self.backgroundColor = backgroundColor
+        self.cornerRadius = cornerRadius
+        self.safeMargin = safeMargin
+    }
+}
+
+public class TextStyleData {
+    public var `default`: TextThemeData
+    public var title: TextThemeData
+    public var subtitle: TextThemeData
+    public var amountLabel: TextThemeData
+    public var system: TextThemeData
+    public var error: TextThemeData
+
+    public init(
+        default: TextThemeData = TextThemeData(),
+        title: TextThemeData = TextThemeData(),
+        subtitle: TextThemeData = TextThemeData(),
+        amountLabel: TextThemeData = TextThemeData(),
+        system: TextThemeData = TextThemeData(),
+        error: TextThemeData = TextThemeData()
+    ) {
+        self.default = `default`
+        self.title = title
+        self.subtitle = subtitle
+        self.amountLabel = amountLabel
+        self.system = system
+        self.error = error
+    }
+}
+
+public class TextThemeData {
+    public var defaultColor: UIColor?
+    public var fontsize: Int?
```

Comment:
[Feedback]
This should be called `fontSize`. Could you update your public interface?